src/radio/radio.js

```diff
@@ -5,5 +5,161 @@ This source code is licensed under the MIT license found in the
 LICENSE file in the root directory of this source tree.
 */
 // @flow
-import {Checkbox as Radio} from '../checkbox/index.js';
+
+import React from 'react';
+
+import {getOverrides} from '../helpers/overrides.js';
+
+import {
+  Root as StyledRoot,
+  Label as StyledLabel,
+  Input as StyledInput,
+  RadioMarkInner as StyledRadioMarkInner,
+  RadioMarkOuter as StyledRadioMarkOuter,
+} from './styled-components.js';
+import type {RadioPropsT, RadioStateT} from './types.js';
+
+function isLabelTopLeft(labelPlacement) {
+  return labelPlacement === 'top' || labelPlacement === 'left';
+}
+
+function isLabelBottomRight(labelPlacement) {
+  return labelPlacement === 'bottom' || labelPlacement === 'right';
+}
+
+class Radio extends React.Component<RadioPropsT, RadioStateT> {
+  static defaultProps: $Shape<RadioPropsT> = {
+    overrides: {},
+    checked: false,
+    disabled: false,
+    autoFocus: false,
+    inputRef: React.createRef(),
+    isError: false,
+    onChange: () => {},
+    onMouseEnter: () => {},
+    onMouseLeave: () => {},
+    onMouseDown: () => {},
+    onMouseUp: () => {},
+    onFocus: () => {},
+    onBlur: () => {},
+  };
+
+  state = {
+    isActive: false,
+    isFocused: this.props.autoFocus || false,
+    isHovered: false,
+  };
+
+  componentDidMount() {
+    if (this.props.autoFocus && this.props.inputRef.current) {
+      this.props.inputRef.current.focus();
+    }
+  }
+
+  onMouseEnter = (e: SyntheticInputEvent<HTMLInputElement>) => {
+    this.setState({isHovered: true});
+    this.props.onMouseEnter && this.props.onMouseEnter(e);
+  };
+
+  onMouseLeave = (e: SyntheticInputEvent<HTMLInputElement>) => {
+    this.setState({isHovered: false, isActive: false});
```

Comment:
Do we need to unset the `isActive` here for the case when mouseUp happens not on the element?

Reply:
looks like the radio mark doesn't implement active states at the moment. also based on figma, it seems like were kind of off here. i'll make a note for us to review here